src/dune/c_rules.mli

```diff
@@ -0,0 +1,13 @@
+open! Stdune
+open Import
+open Dune_file
+
+val build_o_files :
+     Buildable.t
+  -> sctx:Super_context.t
+  -> c_sources:C.Sources.t
+  -> dir:Path.Build.t
+  -> expander:Expander.t
+  -> requires:Lib.L.t Or_exn.t
+  -> dir_contents:Dir_contents.t
+  -> Path.t list
```

Comment:
To make the API a bit better, let's make this return `Path.Build.t list` and have the caller do the transformation. Otherwise, it's not clear that this is returning targets.

Reply:
Yes, done.